analyses/tmb-vaf-longitudinal/04-tmb-unpaired-samples-plot.Rmd

```diff
@@ -0,0 +1,261 @@
+  dplyr::rename(timepoint_cg_n = n) 
+
+# Let's count #samples per cancer groups and timepoints 
+timepoint_cgGFAC_n_df <- df %>% 
+  count(cgGFAC, td_cgGFAC) %>% 
+  dplyr::mutate(tumor_descriptor_cgGFAC_n = glue::glue("{cgGFAC}_{td_cgGFAC}  (N={n})")) %>% 
+  dplyr::rename(timepoint_cgGFAC_n = n) 
+
+# Summarize timepoints per cancer groups
+# We will summarize cancer groups with single timepoints as "Other" 
+timepoint_sum <- df %>%
+  distinct(match_id_DNA, .keep_all = TRUE) %>%
+  group_by(cg_id) %>%
+  arrange(tumor_descriptor) %>%
+  summarise(tumor_descriptor_sum = str_c(tumor_descriptor, collapse = "-")) 
+
+# Create df to use for plots
+df_plot <- df %>% 
+  left_join(timepoint_cg_n_df, by = c("tumor_descriptor", "cg_id")) %>%
+  filter(!tumor_descriptor_cg_n <= 2) %>% # remove if total #samples per timepoint is less than 2
+
+  left_join(timepoint_cgGFAC_n_df, by = c("td_cgGFAC", "cgGFAC")) %>% 
+  filter(!timepoint_cgGFAC_n <= 2) %>% # remove if total #samples per timepoint is less than 2
+
+  left_join(timepoint_sum, by = c("cg_id")) %>%
+  filter(!tumor_descriptor_sum %in% timepoints,
+         !tumor_descriptor_sum %in% c("Diagnosis-Diagnosis", 
+                                      "Diagnosis-Diagnosis-Diagnosis-Diagnosis-Diagnosis",
+                                      "Diagnosis-Diagnosis-Diagnosis-Diagnosis-Diagnosis-Diagnosis",
+                                      "Diagnosis-Diagnosis-Diagnosis-Diagnosis-Diagnosis-Diagnosis-Diagnosis-Diagnosis-Diagnosis"
+                                      )) %>% # remove cancer groups with single timepoints
+  mutate(cg_id_sum = replace_na(cg_id, "Other"),
+         tumor_descriptor = factor(tumor_descriptor),
+         tumor_descriptor = fct_relevel(tumor_descriptor, timepoints))
+``` 
+
+
+## How many patient samples in the unpaired longitudinal data?
+There are `r no_samples` samples in total.
+
+# All unpaired samples
+We are interested in how TMB changes over the time in all PBTA samples (unpaired samples).
+
+```{r define-parameters-for-plots}
+# Read color palette
+palette_df <- readr::read_tsv(palette_file, guess_max = 100000, show_col_types = FALSE) %>% 
+  mutate(tumor_descriptor = color_names)
+
+# Define and order palette
+palette <- palette_df$hex_codes
+names(palette) <- palette_df$tumor_descriptor
+
+# Define label for plots
+Timepoint <- df_plot$tumor_descriptor
+
+# Define ylim
+ylim <- max(df_plot$tmb)
+log10_ylim <- max(df_plot$log10_tmb)
+```
+
+```{r plot-total, fig.width = 10, fig.height = 5, fig.fullwidth = TRUE}
+print(ggplot(df_plot, aes(tumor_descriptor, tmb, color = Timepoint)) + 
+        geom_jitter(width = 0.2, alpha = 0.5, size = 1.5) + 
+        # light guiding line representing 0 exposure
+        geom_hline(yintercept = 0, size = 0.15) + 
+        scale_color_manual(values = palette, 
+                           breaks = sort(names(palette))) +
+        stat_summary(color = "black", size = 0.3) + 
```

Comment:
Since there is no summary function supplied the default `mean_se()` is supplied, but it might be good to define the function in the code so it is noted.

Reply:
Done, see 831e3a44e42a697c29bd3018afc841d7dae075ab